### PR TITLE
Danish fungi2024 merge request

### PR DESCRIPTION
**Pull request by picekl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,119 +1,189 @@
...
-Danish Fungi 2020 - Not Just Another Image Recognition Dataset 
-
-In order to support research in fine-grained plant classification and to allow full reproducibility of our results, we share the Training Logs and Trained scripts.
-- The Images, Checkpoints and Metadata are not included based on size constrains and will be published after the review.
-
-## Training Data
+## Description
+
+Danish Fungi 2020 (DF20) is a new dataset for fine-grained visual categorization. The datasets is 
+constructed from observations submitted to the Atlas of Danish Fungi and is unique in
```
Comment (klarajanouskova) [inline README.md:14]: dataset is / datasets are

File: README.md [hunk 2]
```diff
@@ -1,119 +1,189 @@
...
+DF20 has zero overlap with ImageNet, allowing unbiased comparison of models fine-tuned from publicly 
+available ImageNet checkpoints. The proposed evaluation protocol enables testing the ability to 
+improve classification using metadata – e.g. precise geographic location, habitat, and substrate,
+facilitates classifier calibration testing, and finally allows to study the impact of the device 
+settings on the classification performance.
+
+![Species Similarities and differences](figures/fungi_samples.png)
+
+## News
+- We made a new train/test split based on ObservationIDs where data from the same observation do not occur in the test set.
```
Comment (klarajanouskova) [inline README.md]: maybe we might want to add dates to news so that people now if sth has changed since they last used it

File: README.md [hunk 3]
```diff
@@ -1,119 +1,189 @@
...
+| Danish Fungi 2020        | [LINK [~110GB]](http://ptak.felk.cvut.cz/plants/DanishFungiDataset/DF20-train_val.tar.gz) | [LINK [~6.5GB]](http://ptak.felk.cvut.cz/plants/DanishFungiDataset/DF20-train_val.tar.gz) | [LINK](http://ptak.felk.cvut.cz/plants/DanishFungiDataset/DF20-metadata.zip) |
+| Danish Fungi 2020 – Mini | [LINK [~12.5GB]](http://ptak.felk.cvut.cz/plants/DanishFungiDataset/DF20M-images.tar.gz)  | ---                                                                                     | [LINK](http://ptak.felk.cvut.cz/plants/DanishFungiDataset/DF20M-metadata.zip) |
+| Danish Fungi 2024        | ❗Same as for DF20❗                                                                        | ❗Same as for DF20❗                                                                      | [LINK](http://ptak.felk.cvut.cz/plants/DanishFungiDataset/DanishFungi2024.zip)                                                           |
```
Comment (klarajanouskova) [inline README.md:37]: I thought these were updated?

File: README.md [hunk 4]
```diff
@@ -1,119 +1,189 @@
...
+
+To download the dataset files in CMD, use:
+```
+wget --no-check-certificate URL
+```
 
-Available at -> https://sites.google.com/view/danish-fungi-dataset
+In order to support research in fine-grained plant classification and to allow full reproducibility of our results, we share the training scripts and data tools.
+- Checkpoints are available at [Hugging Face Hub Repository](https://huggingface.co/BVRA).
+- Train and Validation logs are available at [Weights & Biases Workspace](https://wandb.ai/zcu_cv/DanishFungi2023).
```
Comment (klarajanouskova) [inline README.md]: image

File: README.md [hunk 5]
```diff
@@ -1,119 +1,189 @@
...
+
+
+To download the dataset files in CMD, use:
+```
+wget --no-check-certificate URL
+```
 
-Available at -> https://sites.google.com/view/danish-fungi-dataset
+In order to support research in fine-grained plant classification and to allow full reproducibility of our results, we share the training scripts and data tools.
+- Checkpoints are available at [Hugging Face Hub Repository](https://huggingface.co/BVRA).
```
Comment (klarajanouskova) [inline README.md]: maybe it could link directly to https://huggingface.co/collections/BVRA/danish-fungi-2020-66a2228d0f4902df59d549e8

File: README.md [hunk 6]
```diff
@@ -1,119 +1,189 @@
...
-docker run --gpus all -it --rm -v local_dir:container_dir nvcr.io/nvidia/pytorch:21.07-py3
+pip install -r requirements.txt
 ```
-
-2. Install dependencies inside docker container
-
+2. Login to [Weights & Biases](https://wandb.ai/site) to log results [*optional].
 ```
-pip install pandas seaborn timm albumentation tqdm efficientnet_pytorch pretrainedmodels
+import wandb
```
Comment (klarajanouskova) [inline README.md]: I would specify what is done in cmd, what in python, etc. Or maybe give all instructions for cmd, ie 
`wandb login
`

File: README.md [hunk 7]
```diff
@@ -1,119 +1,189 @@
...
+huggingface_hub.login()
 ```
-* Check your paths! 
 
-## Results
+## Training
+For training navigate to `./training` folder.
+To run the training you can use the provided `train.ipynb` notebook or `train.py` CLI.
+In both you have to:
+* Specify valid paths, wandb settings, etc. in **train.ipynb** and run.
```
Comment (klarajanouskova) [inline README.md]: maybe there could be an example like
`python train.py .....`

Comment (klarajanouskova) [reply README.md]: can we be more specific about what exactly they need to set?

Comment (klarajanouskova) [reply README.md]: image
I guess these have names based on how users download it, maybe we can keep the filenames in there and let users specify the path only?

Comment (klarajanouskova) [reply README.md]: Ok the python train.py exasmple is in the notebook, I would just mention it here maybe

File: README.md [hunk 8]
```diff
@@ -1,119 +1,189 @@
...
-| 𐄂  | 𐄂  | 𐄂  | 73.51 | 87.55 | 0.655 |
-| ✔ | 𐄂  | 𐄂  | +1.94 | +1.50 | +0.040  |
-| 𐄂  | ✔ | 𐄂  | +1.23 | +0.95 | +0.020   |
-| 𐄂  | 𐄂  | ✔ | +1.39 | +1.17 | +0.025  |
-| 𐄂  | ✔ | ✔ | +2.47 | +1.98 | +0.042   |
-| ✔ | 𐄂  | ✔ | +3.23 | +2.47 | +0.062   |
-| ✔ | ✔ | 𐄂  | +3.11 | +2.30 | +0.057  | 
-| ✔ | ✔ | ✔ | +3.81 | +2.84 | +0.070 |
+## Exploiting metadata with late-fusion
+To allow further processing model predictions with available metadata we provide a notebook  `./inference/metadata_fusion.ipynb`
```
Comment (klarajanouskova) [inline README.md]: maybe 'updating model predictions by incorporating metadata' or sth like that? Processing model predictions is not clear to me

File: README.md [hunk 9]
```diff
@@ -1,119 +1,189 @@
...
+
+## Results
 
+### CNN Performance Evaluation [**Updated**]
+Classification performance of selected CNN architectures on DF24 and DF24 - Mini.
+All networks share the settings described in Section 6.1 and were trained on 299×299 images.
```
Comment (klarajanouskova) [inline README.md]: Of the paper I guess? I would mention it and maybe also link to it again

File: README.md [hunk 10]
```diff
@@ -1,119 +1,189 @@
...
+| ViT-Base/16      | 69.33    | 85.22   | 57.94 | 76.08    | 88.91    | 66.76 |
+| ViT-Large/16     | 72.20    | 87.46   | 60.23 | 78.81    | 90.64    | 70.25 |
+| ---------------- | ----     | ----    | ----  | ----     | ----     | ----  |
+| Dataset | DF24M    | DF24M   | DF24M | DF24     | DF24     | DF24  |
+
+### Metadata-fusion experiment [**Updated**]
+Performance gains aquired by exploiting the observation metadata, i.e. Habitat (H), Substrate (S), and Month (M).
+Additionally, we provide performance gains based on the ObservationID grouping of calibrated predictions (average over class score).
+
+#### DF24 - ViT-Large/16 with image size 384×384. [**Updated**]
```
Comment (klarajanouskova) [inline README.md:153]: what does the update refer to?

File: inference/utils/DanishFungiDataset.py [hunk 11]
```diff
@@ -0,0 +1,138 @@
+import cv2
+import pandas as pd
+from albumentations import Compose, Normalize, Resize
+from albumentations.pytorch import ToTensorV2
+from torch.utils.data import Dataset
+
+
+class DanishFungiDataset(Dataset):
```
Comment (klarajanouskova) [inline inference/utils/DanishFungiDataset.py:8]: why not use the fgvc dataset class since they need to install it anyway? It is nice for people who only want the dataset without training etc. thoug.

File: inference/utils/DanishFungiDataset.py [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+
+        selected_feature_values = {
+            feature: self.df[feature].iloc[idx] for feature in self.extra_features
+        }
+
+        return image, label, file_path, selected_feature_values
+
+
+def get_transforms(
+    model_mean: list[float], model_std: list[float], image_size: tuple[int, int]
```
Comment (klarajanouskova) [inline inference/utils/DanishFungiDataset.py:117]: why is this a function of the model?

File: README.md [hunk 13]
```diff
@@ -1,119 +1,189 @@
...
+| ---------------- |----------|---------|-------|----------|----------|-------|
+| EfficientNet-B0  | 63.79    | 81.60   | 51.22 | 70.16    | 85.00    | 59.34 |
+| EfficientNet-B3  | 65.14    | 82.46   | 52.55 | 72.47    | 86.63    | 62.31 |
+| SE-ResNeXt-101   | 68.06    | 84.00   | 56.22 | 74.83    | 88.13    | 65.32 |
+| ViT-Base/16      | 69.33    | 85.22   | 57.94 | 76.08    | 88.91    | 66.76 |
+| ViT-Large/16     | 72.20    | 87.46   | 60.23 | 78.81    | 90.64    | 70.25 |
+| ---------------- | ----     | ----    | ----  | ----     | ----     | ----  |
+| Dataset | DF24M    | DF24M   | DF24M | DF24     | DF24     | DF24  |
+
+### Metadata-fusion experiment [**Updated**]
```
Comment (klarajanouskova) [inline README.md:147]: is there a reference?
